src/SFML/Graphics/Shader.cpp

```diff
@@ -156,11 +210,50 @@ Shader::CurrentTextureType Shader::CurrentTexture;
 
 
 ////////////////////////////////////////////////////////////
+struct Shader::UniformBinder : private sf::NonCopyable
+{
+    ////////////////////////////////////////////////////////////
+    /// \brief Constructor: set up state before uniform is set
+    ///
+    ////////////////////////////////////////////////////////////
+    UniformBinder(Shader* shader, const std::string& name) :
+    program(0),
+    location(-1)
+    {
+        if (shader->m_shaderProgram)
```

Comment:
You might have seen this coming: Why not define in an anonymous namespace and just inherit from `GlResource` in there? :stuck_out_tongue: